instrumentation/opentelemetry/init.go

```diff
@@ -101,6 +113,36 @@ func makeExporterFactory(cfg *config.AgentConfig) func() (sdktrace.SpanExporter,
 	}
 }
 
+func createTLSConfig(cfg *config.AgentConfig) *tls.Config {
+	tlsConfig := &tls.Config{}
+	tlsConfig.InsecureSkipVerify = !cfg.GetReporting().GetSecure().GetValue()
+	tlsConfig.RootCAs = createCaCertPoolFromFile(cfg)
+
+	return tlsConfig
+}
+
+// createCaCertPoolFromFile creates a CA Cert Pool from a file path containing
+// a raw CA certificate to verify a server certificate. The file path is the
+// reporting.cert_file config value.
+func createCaCertPoolFromFile(cfg *config.AgentConfig) *x509.CertPool {
```

Comment:
This function does not need to be aware of the whole `*config.AgentConfig`, we should just use `*config.CertFile`, the less the better.

Reply:
done

Reply:
How about covering this function with tests?